src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateMatchInfoExtensions.cs

```diff
@@ -16,14 +16,30 @@ public static bool HasNameMismatch(this ITemplateMatchInfo templateMatchInfo)
         public static bool HasParameterMismatch(this ITemplateMatchInfo templateMatchInfo)
         {
             return templateMatchInfo.MatchDisposition.Any(x => x.Location == MatchLocation.OtherParameter
-                                       && (x.Kind != MatchKind.Exact && x.Kind != MatchKind.AmbiguousParameterValue && x.Kind != MatchKind.SingleStartsWith));
+                                       && x.Kind != MatchKind.Exact && x.Kind != MatchKind.AmbiguousParameterValue && x.Kind != MatchKind.SingleStartsWith);
         }
 
         public static bool HasContextMismatch(this ITemplateMatchInfo templateMatchInfo)
         {
             return templateMatchInfo.MatchDisposition.Any(x => x.Location == MatchLocation.Context && x.Kind == MatchKind.Mismatch);
         }
 
+        public static bool HasLanguageMismatch(this ITemplateMatchInfo templateMatchInfo)
```

Comment:
minor: can you confirm the access modifiers for this class & its' members must be public?

Reply:
I agree that it would be better to have them internal and visible to unit test assembly, however none of the classes is using that approach for a moment. I would suggest a separate task to investigate it and fix all public classes that used for unit tests only / other projects.